src/map/zone.cpp

```diff
@@ -1126,6 +1144,34 @@ void CZone::CharZoneOut(CCharEntity* PChar)
     moduleutils::OnCharZoneOut(PChar);
     luautils::OnZoneOut(PChar);
 
+    if (PChar->m_LevelRestriction != 0)
+    {
+        if (PChar->PParty)
+        {
+            if (PChar->PParty->GetSyncTarget() == PChar || PChar->PParty->GetLeader() == PChar)
+            {
+                PChar->PParty->SetSyncTarget("", 551);
+            }
+            if (PChar->PParty->GetSyncTarget() != nullptr)
+            {
+                uint8 count = 0;
+                for (uint32 i = 0; i < PChar->PParty->members.size(); ++i)
+                {
+                    if (PChar->PParty->members.at(i) != PChar && PChar->PParty->members.at(i)->getZone() == PChar->PParty->GetSyncTarget()->getZone())
+                    {
+                        count++;
+                    }
+                }
+                if (count < 2) // 3, because one is zoning out - thus at least 2 will be left
+                {
+                    PChar->PParty->SetSyncTarget("", 552);
```

Comment:
Inside of `SetSyncTarget`, this magic 552 is being cast to `static_cast<MsgStd>(message)));`. While you're around this bit of code, can you add new entries to `MsgStd` and then use them in all the places `SetSyncTarget` is used, and then remove that cast?

It'll end up looking like:
```cpp
// Callsite
PChar->PParty->SetSyncTarget("", MsgStd::TheNewEnumWhatever);

// Function sig
void CParty::SetSyncTarget(const std::string& MemberName, MsgStd message)

// In the function (old)
member->pushPacket(new CMessageStandardPacket(PChar->GetMLevel(), 0, 0, 0, static_cast<MsgStd>(message)));

// new
member->pushPacket(new CMessageStandardPacket(PChar->GetMLevel(), 0, 0, 0, message));
```

Reply:
Thanks for the guidance here! 

Went ahead and added the existing magic numbers to MsgStd, also found a few more that I figured would be good to add that aren't currently implemented (but figured it would be better to have them at the ready while I was there, let me know if those unused should be removed)